### ISSUE TITLE
getGenericSuperclass fails on a type parametrized with Validation (MalformedParameterizedTypeException).

### ISSUE THREAD

**Issue by dpolshin** [scalaz/scalaz#828]
hello,
i'm having issue with reflection failing on types parametrized with scalaz.Validation.
here's a code snippet that fails on 7.1.0 but works fine on 7.0.6.

build.sbt:
```scala
scalaVersion := "2.11.2"

sbtVersion := "0.13.6"

libraryDependencies ++= Seq(
  "org.scalaz" %% "scalaz-core" % "7.1.0"
)
```

Test.scala:
```scala
import _root_.scalaz.Validation

abstract class AbstractClass[T <: AnyRef]

class SomeClass extends AbstractClass[Validation[List[String], Object]]

object Test extends App {
  classOf[SomeClass].getGenericSuperclass
}
```

results in:
```
> run
[info] Running Test
[error] (run-main-6) java.lang.reflect.MalformedParameterizedTypeException
java.lang.reflect.MalformedParameterizedTypeException
        at sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl.validateConstructorArguments(ParameterizedTypeImpl.java:60)
        at sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl.<init>(ParameterizedTypeImpl.java:53)
        at sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl.make(ParameterizedTypeImpl.java:95)
        at sun.reflect.generics.factory.CoreReflectionFactory.makeParameterizedType(CoreReflectionFactory.java:105)
        at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:140)
        at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
        at sun.reflect.generics.visitor.Reifier.reifyTypeArguments(Reifier.java:68)
        at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:138)
        at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
        at sun.reflect.generics.repository.ClassRepository.getSuperclass(ClassRepository.java:84)
        at java.lang.Class.getGenericSuperclass(Class.java:696)
        at Test$.delayedEndpoint$Test$1(Test.scala:9)
        at Test$delayedInit$body.apply(Test.scala:8)
        at scala.Function0$class.apply$mcV$sp(Function0.scala:40)
        at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
        at scala.App$$anonfun$main$1.apply(App.scala:76)
        at scala.App$$anonfun$main$1.apply(App.scala:76)
        at scala.collection.immutable.List.foreach(List.scala:381)
        at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
        at scala.App$class.main(App.scala:76)
        at Test$.main(Test.scala:8)
        at Test.main(Test.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
[trace] Stack trace suppressed: run last compile:run for the full output.
java.lang.RuntimeException: Nonzero exit code: 1
        at scala.sys.package$.error(package.scala:27)
[trace] Stack trace suppressed: run last compile:run for the full output.
[error] (compile:run) Nonzero exit code: 1
[error] Total time: 0 s, completed Oct 13, 2014 2:15:44 PM
```

**Comment by Milyardo** [comment]
The underlying cause, [SI-7932](https://issues.scala-lang.org/browse/SI-7932) was reported to be fixed in 2.11.0-RC1, given that resolves #571 `-Yno-generic-signatures` should only be enabled when building for 2.10.x.

**Comment by Milyardo** [comment]
This issue is fixed by #932.
